modules/apps/foundation/roles/roles-admin-test/src/testIntegration/java/com/liferay/roles/admin/web/util/test/RoleAssigneesUtilTest.java

```diff
@@ -0,0 +1,294 @@
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.roles.admin.web.util.RoleAssigneesUtil;
+
+import javax.servlet.http.HttpServletRequest;
+
+import junit.framework.Assert;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Drew Brokke
+ */
+@RunWith(Arquillian.class)
+public class RoleAssigneesUtilTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(), TransactionalTestRule.INSTANCE);
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_regularRole = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+		User regularRoleUser = UserTestUtil.addUser();
+		Group regularRoleSite = GroupTestUtil.addGroup();
+		Organization regularRoleOrganization =
+			OrganizationTestUtil.addOrganization();
+		UserGroup regularRoleUserGroup = UserGroupTestUtil.addUserGroup();
+
+		RoleLocalServiceUtil.addUserRole(
+			regularRoleUser.getUserId(), _regularRole);
+		RoleLocalServiceUtil.addGroupRole(
+			regularRoleSite.getGroupId(), _regularRole);
+		RoleLocalServiceUtil.addGroupRole(
+			regularRoleOrganization.getGroupId(), _regularRole);
+		RoleLocalServiceUtil.addGroupRole(
+			regularRoleUserGroup.getGroupId(), _regularRole);
+
+		_siteRole = RoleTestUtil.addRole(RoleConstants.TYPE_SITE);
+
+		Group testSite = GroupTestUtil.addGroup();
+		User siteRoleUser = UserTestUtil.addUser();
+		UserGroup siteRoleUserGroup = UserGroupTestUtil.addUserGroup();
+
+		GroupLocalServiceUtil.addUserGroup(siteRoleUser.getUserId(), testSite);
+		GroupLocalServiceUtil.addUserGroupGroup(
+			siteRoleUserGroup.getUserGroupId(), testSite);
+
+		long[] siteRoleIdArray = new long[] {_siteRole.getRoleId()};
+
+		UserGroupRoleLocalServiceUtil.addUserGroupRoles(
+			siteRoleUser.getUserId(), testSite.getGroupId(), siteRoleIdArray);
+		UserGroupGroupRoleLocalServiceUtil.addUserGroupGroupRoles(
+			siteRoleUserGroup.getGroupId(), testSite.getGroupId(),
+			siteRoleIdArray);
+
+		_organizationRole = RoleTestUtil.addRole(
+			RoleConstants.TYPE_ORGANIZATION);
+
+		Organization testOrganization = OrganizationTestUtil.addOrganization();
+		User organizationRoleUser = UserTestUtil.addUser();
+
+		OrganizationLocalServiceUtil.addUserOrganization(
+			organizationRoleUser.getUserId(), testOrganization);
+
+		UserGroupRoleLocalServiceUtil.addUserGroupRoles(
+			organizationRoleUser.getUserId(), testOrganization.getGroupId(),
+			new long[] {_organizationRole.getRoleId()});
+	}
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		RoleLocalServiceUtil.deleteRole(_regularRole);
+		RoleLocalServiceUtil.deleteRole(_siteRole);
+		RoleLocalServiceUtil.deleteRole(_organizationRole);
```

Comment:
I just realized that this teardown is insufficient, since I added other users, groups, and organization in the setup method :/

I can add it when I get back.